Shopify/shopify-cli
# [Feature]: Changed interactive apps list scope for extension create|register|connect commands

### WHY are these changes introduced?
Fixes: https://github.com/Shopify/shopify-cli-planning/issues/90

When we show you apps to potentially connect your extension to via any of the `extension create / register / connect` commands, we show you all the apps that you have access to across all your partner accounts, not only the apps that are in your currently logged in partner account.

That behavior's a bit odd from an auth / permissions angle and we should fix that.

Related:
https://github.com/Shopify/shopify-cli/issues/1839 for `extension create`
https://github.com/Shopify/shopify-cli/issues/1899 for `extension connect`
https://github.com/Shopify/shopify-cli/issues/2234

### WHAT is this pull request doing?
- Updated `Task::GetApps.call` to return a list of apps for logged in organization
- Created new method `ShopifyCLI::PartnersAPI::Organizations.fetch_with_app` that returns data about organization and their apps 
- Renamed `ShopifyCLI::PartnersAPI::Organizations.fetch_with_app` to `ShopifyCLI::PartnersAPI::Organizations.fetch_all_with_apps` for better context fit

### How to test your changes?
Pre-requisites: Have 2 or more partner accounts. .
1. Test `extension create`
- Log out from all partner accounts.
- Login into one partner account `shopify login`
- From your application directory run `shopify extension create`
You see only the apps that are in your currently logged in partner account.

2. Test `extension register`
- Log out from all partner accounts.
- Login into one partner account `shopify login`
- From your extension directory run `shopify extension register`



### Post-release steps



### Update checklist
